### PR TITLE
Fix eigenvector testing and HITS testing discrepancies

### LINKED ISSUE #3971: [BUG]: `plc.eigenvector_centrality` fails to converge on simple graph
### Version

23.12 dev

### Which installation method(s) does this occur on?

Source

### Describe the bug.

`plc.eigenvector_centrality` fails to converge on a simple graph. NetworkX converges in a few iterations.

The adjacency matrix of this graph is:
```
[[0 1 0]
 [1 0 1]
 [0 1 0]]
```
The expected result is `[0.5, 0.5**0.5, 0.5]`.

### Minimum reproducible example

```shell
import cupy as cp
import numpy as np
import pylibcugraph as plc

src_indices = cp.array([0, 1, 1, 2], dtype=np.int32)
dst_indices = cp.array([1, 0, 2, 1], dtype=np.int32)

G = plc.SGGraph(
    resource_handle=plc.ResourceHandle(),
    graph_properties=plc.GraphProperties(
        is_multigraph=False,
        is_symmetric=True,
    ),
    src_or_offset_array=src_indices,
    dst_or_index_array=dst_indices,
    weight_array=None,  # Also fails with array of all 1s
    store_transposed=True,
    renumber=False,
    do_expensive_check=False,
)
# cpp/src/centrality/eigenvector_centrality_impl.cuh line=151: Eigenvector Centrality failed to converge.
node_ids, values = plc.eigenvector_centrality(
    resource_handle=plc.ResourceHandle(),
    graph=G,
    epsilon=0.1,  # This is really high!
    max_iterations=1000,  # This is super high! We should converge in a few steps
    do_expensive_check=False,
)
```
```


### Relevant log output

_No response_

### Environment details

_No response_

### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open bugs](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### LINKED ISSUE #3972: [BUG]: `plc.hits` does not raise after not converging after reaching max iterations
### Version

23.12 dev

### Which installation method(s) does this occur on?

_No response_

### Describe the bug.

Unlike other power iteration methods, HITS does not raise upon reaching the maximum number of iterations without converging.

See here that it does not check the convergence criteria and raise if necessary:
https://github.com/rapidsai/cugraph/blob/f4bcdc2667a15e2e5031987550322a9d29d8f713/cpp/src/link_analysis/hits_impl.cuh#L176-L192

For an example of expected behavior, here's Katz centrality:
https://github.com/rapidsai/cugraph/blob/f4bcdc2667a15e2e5031987550322a9d29d8f713/cpp/src/centrality/katz_centrality_impl.cuh#L163-L168

### Minimum reproducible example

```shell
Call with `max_iterations=1`.
```

### Relevant log output

_No response_

### Environment details

_No response_

### Other/Misc.

Not having this behavior makes it difficult for `nx-cugraph` to match NetworkX.

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open bugs](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by ChuckHastings**
Address a couple of bugs discovered in nx-cugraph testing.

HITS should raise an exception if it fails to converge.

Eigenvector computation was not quite right, nx-cugraph testing discovered an edge condition where it was obvious.

Closes #3971
Closes #3972

### CODE REVIEW

File: cpp/tests/c_api/eigenvector_centrality_test.c [hunk 1]
```diff
@@ -109,11 +109,30 @@ int test_eigenvector_centrality()
     h_src, h_dst, h_wgt, h_result, num_vertices, num_edges, TRUE, epsilon, max_iterations);
 }
 
+int test_eigenvector_centrality_3971()
+{
+  size_t num_edges    = 4;
```
Comment (naimnv) [inline cpp/tests/c_api/eigenvector_centrality_test.c:114]: Wouldn't it be 8? IIRC it's the length of wight array.

**Review by naimnv — APPROVED** [review]

**Review by seunghwak — APPROVED** [review]
LGTM

**Review by rlratzel — APPROVED** [review]
